### PR TITLE
[Docs] Feature-store + Datastore: consolidate Apache Kafka details, cleanups in datastores

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/feature-store/sources-targets.md [hunk 1]
```diff
@@ -17,101 +16,57 @@ You can also create a custom `source` to access various databases or data source
...
-| {py:meth}`~mlrun.datastore.StreamSource`                           |Event-based. Sets the stream source for the flow. If the stream doesn’t exist it creates it. | Y      | N     | N      |
+| {py:meth}`~mlrun.datastore.ParquetSource`                                                      | Batch. Reads the Parquet file/dir as the input source for a flow.     | Y      | Y     | Y      |
+| [S3/Azure source](#s3-azure-source)                                                            | Batch.                                                                 |       |      |       |
+| {py:meth}`~mlrun.datastore.HttpSource`                                                          |Event-based. Sets the HTTP-endpoint source for the flow.    | Y      | N     | N      |
+| [Kafka source](#kafka-source)                                                  |Event-based. Sets a Kafka source for the flow (supports both Apache and Confluence Kafka).| Y      | N     | N      |
+| [Confluent Kafka source](#confluent-kafka-source)                                            |Event-based. Sets the kafka source for the flow.          | Y      | N     | N      |
```
Comment (assaf758) [inline docs/feature-store/sources-targets.md]: There is only one class - KafkaSource (and you are mentioning Confluent in the above line).
So this line should probably be removed.

File: docs/feature-store/sources-targets.md [hunk 2]
```diff
@@ -200,6 +204,14 @@ It supports low latency data retrieval based on key access, making it ideal for
 
 The combination of a NoSQL target with the storey engine does not support features of type string with a value containing both quote (') and double-quote (").
 
+## Redis target
```
Comment (assaf758) [inline docs/feature-store/sources-targets.md]: both "Redis target" and "RedisNoSql target"?
Also I suggest we consistently:
1. use the class name as the section name - what do you think? So for example - RedisNoSqlTarget
2. First explain the class parameters, then describe the associated datastore profile

File: docs/store/datastore.md [hunk 3]
```diff
@@ -310,11 +319,14 @@ ParquetTarget(path="ds://test_profile/aws_bucket/path/to/parquet.pq")
...
 
-## See also
-- {py:class}`~mlrun.projects.MlrunProject.list_datastore_profiles` 
-- {py:class}`~mlrun.projects.MlrunProject.get_datastore_profile`
-- {py:class}`~mlrun.datastore.datastore_profile.register_temporary_client_datastore_profile`
-- {py:class}`~mlrun.projects.MlrunProject.delete_datastore_profile`
-
-The methods `get_datastore_profile()` and `list_datastore_profiles()` only return public information about 
-the profiles. Access to private attributes is restricted to applications running in Kubernetes pods.
+## Adding a data store profile
```
Comment (assaf758) [inline docs/store/datastore.md]: Checking the formal docs there is not info for developers.
So keeping with this, I think "adding data store profile" (and adding datastore) should go to a separate directory (docs/internal/datastores ?), and being pointed from CONTRIBUTING.md

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: docs/store/datastore.md (hunk 3 above)
Comment (jillnogold) [reply docs/store/datastore.md]: Developers' docs stay in the build since there is no other home for them

Comment (assaf758) [reply docs/store/datastore.md]: well, i'm not in agreement with that - it's the same as CONTRIBUTING.md which is not mentioned in the formal docs.
Information for developers shouldn't be mixed into the user documentation - it's for different audience with different technical knowledge and in my opinion it ruins the quality of the docs

File: docs/store/datastore.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ Data stores are referred to using the schema prefix (e.g. `s3://my-bucket/path`)
 * **store** &mdash; MLRun versioned artifacts [(see Artifacts)](./artifacts.html), format: `store://artifacts/<project>/<artifact-name>[:tag]`
 * **memory** &mdash; in memory data registry for passing data within the same process, format `memory://key`, use `mlrun.datastore.set_in_memory_item(key, value)` 
    to register in memory data items (byte buffers or DataFrames). (Not supported by all Spark runtimes)
+* Additional data stores, for example MongoDB, can easily be added by extending the `DataStore` class. See [Adding a data store profile](#adding-a-data-store-profile)
```
Comment (assaf758) [inline docs/store/datastore.md:19]: 1. Remove: "See "Adding a data store profile"
2. Add oss - Alibaba Cloud Object Storage Service
3. Sort by the order it appears in the chapter
4. Create link to the relevant section

**Review by assaf758 — APPROVED** [review]

**Pull request merged**